src/expr/emptybag.h

```diff
@@ -0,0 +1,65 @@
+/*********************                                                        */
+/*! \file emptybag.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Tim King, Kshitij Bansal, Morgan Deters
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief [[ Add one-line brief description here ]]
+ **
+ ** [[ Add lengthier description here ]]
+ ** \todo document this file
+ **/
+
+#include "cvc4_public.h"
+
+#ifndef CVC4__EMPTY_BAG_H
+#define CVC4__EMPTY_BAG_H
+
+#include <iosfwd>
+#include <memory>
+
+namespace CVC4 {
+
+class TypeNode;
+
+class CVC4_PUBLIC EmptyBag
+{
+ public:
+  /**
+   * Constructs an emptybag of the specified type. Note that the argument
+   * is the type of the bag itself, NOT the type of the elements.
+   */
+  EmptyBag(const TypeNode& bagType);
+  ~EmptyBag();
+  EmptyBag(const EmptyBag& other);
+  EmptyBag& operator=(const EmptyBag& other);
+
+  const TypeNode& getType() const;
+  bool operator==(const EmptyBag& es) const;
+  bool operator!=(const EmptyBag& es) const;
+  bool operator<(const EmptyBag& es) const;
+  bool operator<=(const EmptyBag& es) const;
+  bool operator>(const EmptyBag& es) const;
+  bool operator>=(const EmptyBag& es) const;
+
+ private:
+  EmptyBag();
+
+  std::unique_ptr<TypeNode> d_type;
```

Comment:
Can you document this?  In particular, state that it is the element type (not the set type).

Reply:
Actually it is the bag type (not the element type)

Reply:
Done